docs/guide/teal-debugger.md

```diff
@@ -0,0 +1,191 @@
+---
+layout: splash
+---
+
+# About TEAL Debugger
+
+Algorand provides the `tealdbg` command-line tool to launch an interactive session where the smart contract can be examined as the contract is being evaluated. The debugger supports both stateful and stateless smart contracts. You can debug individual transactions or atomic transfers that contain multiple, grouped transactions. The debugger supports setting the specific context for debugging purposes, including transactions, round number, latest timestamp, balance records, etc. The debugger runs either local programs or accepts HTTP connections from remote evaluators configured to run with a remote debugger hook.
+
+# Using teal debugger with algob
+
+Setting up a `--dryrun-dump` of transaction data (via `goal` or SDK) could be a lengthy process, especially when using a transaction group. `Algob` provides an easy way to use debugger: by just supplying the transactions as an input to the `TealDbg` method (same transaction data that we supply to [executeTransaction](https://algobuilder.dev/api/algob/modules.html#executetransaction) to execute same transaction on network).
+
+## Using Dryrun for Debugging a TEAL Program in an algob script
+
+Algob provides the functionality to do a test run of a TEAL smart contract. This option is useful to not only capture the transaction in the output file but the associated state of a smart contract. This allows testing of the TEAL logic in a dry run state which shows how the TEAL is processed and approved or rejected.
+
+The dryrun response to this REST API includes disassembly, logic signature messages with PASS/REJECT, a signature trace, app call messages, and an app call trace.
+
+NOTE: The dryrun REST API is only available on a node if it has been enabled in the node's configuration (`EnableDeveloperAPI` = true).
+Example:
+```js
+// txnParams are the input transactions
+const debugger = new Tealdbg(deployer, txnParam);
+await debugger.dryRunResponse('dryrun.json');
+```
+
+The DryRun Response will be dumped to `assets/dryrun.json`.
```

Comment:
shouldn't this be in artifacts/cache ?

Reply:
no, I don't think it's an intermediate file, user will inspect the response after execution. Should this be in artifacts/cache?

Reply:
yes I think so, because user doesn't create this. So it should be a part of artifacts

Reply:
ok, will update while adding support for stateful. Let's dump in `artifacts/dryrun/**`. But this means these files will be cleared by `algob clean`.